test/distributed/test_c10d_nccl.py

```diff
@@ -4257,6 +4258,127 @@ def gather_trace():
+            return
+        pg = self._create_process_group_nccl()
+        if timing_enabled:
+            pg._enable_collectives_timing()
+
+        num_coalesced_ops = 20
+        ops_per_coalesce = len(op_sizes_per_coalesce)
+        for i in range(num_coalesced_ops):
+            ops = []
+            for input_sizes in op_sizes_per_coalesce:
+                tensor = torch.zeros(input_sizes).to(self.local_device)
+                if self.rank == 0:
+                    ops.append(dist.P2POp(dist.irecv, tensor, 1))
+                elif self.rank == 1:
+                    tensor *= 2
+                    ops.append(dist.P2POp(dist.isend, tensor, 0))
+
+            dist.batch_isend_irecv(ops).pop().wait()
+
+        torch.cuda.synchronize()
+
+        if timing_enabled:
+            # wait for watchdog thread to process the queue of works
+            time.sleep(1)
+
+        t = pickle.loads(torch._C._distributed_c10d._dump_nccl_trace())
+        self.assertEqual(len(t['entries']), num_coalesced_ops * (ops_per_coalesce + 1))
+        for seq in range(num_coalesced_ops):
+            first_op = seq * (ops_per_coalesce + 1)
+            coalesced_op = first_op + ops_per_coalesce
+            expected_seq = seq + 1
+            for p2p_op_idx, input_sizes in zip(range(first_op, coalesced_op, 1), op_sizes_per_coalesce):
+                # the indivudal ops inside the coalescing group the individual op metadata,
+                # but not the timing info coming from the actual coalesced kernel
+                profiling_name = 'nccl:recv 0<-1' if self.rank == 0 else 'nccl:send 1->0'
+                self.assertEqual(t['entries'][p2p_op_idx]['profiling_name'], profiling_name)
+                self.assertEqual(t['entries'][p2p_op_idx]['seq_id'], expected_seq)
+                self.assertEqual(t['entries'][p2p_op_idx]['input_sizes'], [input_sizes])
+                self.assertEqual(t['entries'][p2p_op_idx]['output_sizes'], [input_sizes])
+                # duration doesn't get tagged onto individual ops yet, nor is their state updated
+                self.assertEqual(t['entries'][p2p_op_idx]['state'], 'scheduled')
+                self.assertTrue('duration_ms' not in t['entries'][p2p_op_idx])
+
+            # the coalesced op has no metadata but indicates that coalescing was used,
+            # and accurately reflects the timing and state info for the whole group
+            self.assertEqual(t['entries'][coalesced_op]['profiling_name'], 'nccl:coalesced')
+            self.assertEqual(t['entries'][coalesced_op]['seq_id'], expected_seq)
+            self.assertEqual(t['entries'][coalesced_op]['state'], 'completed')
+            self.assertEqual(t['entries'][coalesced_op]['input_sizes'], [])
+            self.assertEqual(t['entries'][coalesced_op]['output_sizes'], [])
+
+            if timing_enabled:
+                duration = t['entries'][coalesced_op]['duration_ms']
+                self.assertTrue(0.001 < duration < 10000, duration)
```

Comment:
this doesn't come from anything. I just made it up.  It is just supposed to be 'sufficiently wide bounds that it will always pass if a real collective ran, but will catch a zero value that could happen if no collective ran, or a garbage value if we emitted some kind of corrupted data